bookings/views.py

```diff
@@ -1,3 +1,84 @@
-from django.shortcuts import render
+import json
+from datetime         import datetime
 
-# Create your views here.
+from django.views     import View
+from django.http      import JsonResponse
+from django.db.models import Q
+
+from .models          import Booking
+from products.models  import Product
+from users.utils      import login_decorator
+
+class BookingView(View):
+    @login_decorator
+    def post(self, request, product_id):
+        try:
+            data        = json.loads(request.body)
+            
+            check_in    = data.get("checkin")
+            check_out   = data.get("checkout")
+            head_count  = data.get("count")
+            
+            if not check_in or not check_out or not head_count:
+                raise KeyError
+            
+            first  = datetime.strptime(check_in, "%Y-%m-%d")
+            second = datetime.strptime(check_out, "%Y-%m-%d")
```

Comment:
`start_date`, `end_date`가 더 명확할 것 같네요. (자주 사용되는 변수명이니 기억하면 좋을 것 같아요~)

```suggestion
            start_date  = datetime.strptime(check_in, "%Y-%m-%d")
            end_date = datetime.strptime(check_out, "%Y-%m-%d")
```